tests/features/records-tags.spec.ts

```diff
@@ -54,6 +58,1902 @@ export function testRecordsTags(): void {
     });
 
     describe('RecordsWrite with tags', () => {
+      describe('protocol rules', () => {
```

Comment:
Can we add a sanity test (I did not see it) for making sure the definition is rejected when an invalid JSON schema was given for the `$tags`.

e.g.  a string is assigned to `maximum` instead of a number.

Reply:
GREAT catch! It actually isn't validating that.

Let me see if there is a better way to validate that instead of explicitly writing those rules in the already large JSON schema definition for tags.